### ISSUE TITLE
Error "pending queries not implemented" when deploying with Truffle/Buidler

### ISSUE THREAD

**Issue by okwme** [cosmos/ethermint#156]
__System info:__Ethermint  6eef37b0c6adf48582d1334e0766d40dacf76b02

__Steps to reproduce:__

1. Using https://github.com/clovers-network/clovers-contracts to try suite with ethermint
2. Add ethermint account private key to .env as `ETHERMINT_KEY`
3. run command `> npx buidler test --no-compile --network ethermint` after starting up ethermint
4. fails with error:
```
Error: Returned error: invalid argument 1: pending queries not implemented\n    at Object.ErrorResponse (/home/billy/GitHub.com/clovers-network/clovers-contracts/node_modules/web3-core-helpers/src/errors.js:29:16)\n    at /home/billy/GitHub.com/clovers-network/clovers-contracts/node_modules/web3-core-requestmanager/src/index.js:144:36\n    at cb (util.js:353:39)\n    at process._tickCallback (internal/process/next_tick.js:63:19)
```

__Expected behavior:__ Be able to deploy and test contracts like possible with ganache-cli

__Actual behavior:__ Error occurs

__Additional info:__ I dug into the line inside of `web3-core-requestmanager` and found the error occurs with this rpc query:
```
{
	"jsonrpc": "2.0",
	"id": 20,
	"method": "eth_sendTransaction",
	"params": [
		{
			"from": "0xa4fc1e18f38debb934a195435982942c56aa2dd1",
			"data": "0x60806040523480156200001157600080fd5b506040516200325238038062003252833981016040528051602082015190820191018181620000697f01ffc9a7000000000000000000000000000000000000000000000000000000006401000000006200019d810204565b6200009d7f80ac58cd000000000000000000000000000000000000000000000000000000006401000000006200019d810204565b620000d17f4f558e79000000000000000000000000000000000000000000000000000000006401000000006200019d810204565b8151620000e69060059060208501906200020a565b508051620000fc9060069060208401906200020a565b50620001317f780e9d63000000000000000000000000000000000000000000000000000000006401000000006200019d810204565b620001657f5b5e139f000000000000000000000000000000000000000000000000000000006401000000006200019d810204565b5050336000818152600c60205260409020805460ff19166001179055600d8054600160a060020a031916909117905550620002af9050565b7fffffffff000000000000000000000000000000000000000000000000000000008082161415620001cd57600080fd5b7fffffffff00000000000000000000000000000000000000000000000000000000166000908152602081905260409020805460ff19166001179055565b828054600181600116156101000203166002900490600052602060002090601f016020900481019282601f106200024d57805160ff19168380011785556200027d565b828001600101855582156200027d579182015b828111156200027d57825182559160200191906001019062000260565b506200028b9291506200028f565b5090565b620002ac91905b808211156200028b576000815560010162000296565b90565b612f9380620002bf6000396000f3006080604052600436106102875763ffffffff7c010000000000000000000000000000000000000000000000000000000060003504166301ffc9a7811461028957806303a53d6b146102d7578063048e62ca14610301578063049526311461033957806306fdde0314610351578063081812fc146103db578063095ea7b31461041c5780630b6283951461044d5780630e8310d61461049057806312414198146104a85780631732e941146104bd57806318160ddd146104fd57806319fa8f50146105125780631c4b774b1461055c5780631cc1472d1461057457806323b872dd146105a557806324d7806c146105dc5780632f745c591461060a57806340c10f191461063b57806342842e0e1461066c578063429b62e5146106a35780634e485f88146106d15780634f558e79146106ff5780634f6ccce71461071757806350a98a641461072f5780635be7cc16146107445780635c60da1b1461077257806362f9acf9146107875780636352211e146107da57806364e779b1146107f25780636fce9d881461080a578063708fa7f71461083857806370a0823114610878578063715018a6146108a657806376e4a87f146108bb578063785d8d71146108d85780638810ee8b1461092057806388b6e7a514610a5a5780638a56a22614610a6f5780638b1b2ba814610a9d5780638da5cb5b14610ac45780639564e18014610ad957806395d89b4114610b07578063a22cb46514610b1c578063a47bd49614610b4f578063a89b55da14610b6a578063b88d4fde14610b7f578063c53b4bf914610bfb578063c863bac114610c13578063c87b56dd14610c4b578063dc3f775814610c63578063e3100d1e14610c7e578063e985e9c514610c99578063f2fde38b14610ccd575b005b34801561029557600080fd5b506102c37fffffffff0000000000000000000000000000000000000000000000000000000060043516610cfb565b604080519115158252519081900360200190f35b3480156102e357600080fd5b506102ef600435610d32565b60408051918252519081900360200190f35b34801561030d57600080fd5b506102c373ffffffffffffffffffffffffffffffffffffffff6004358116906024359060443516610d47565b34801561034557600080fd5b506102c3600435610f29565b34801561035d57600080fd5b50610366610f3e565b6040805160208082528351818301528351919283929083019185019080838360005b838110156103a0578181015183820152602001610388565b50505050905090810190601f1680156103cd5780820380516001836020036101000a031916815260200191505b509250505060405180910390f35b3480156103e757600080fd5b506103f3600435610ff3565b6040805173ffffffffffffffffffffffffffffffffffffffff9092168252519081900360200190f35b34801561042857600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff6004351660243561101b565b34801561045957600080fd5b5060408051808201825261028791600480359236926064919060249060029083908390808284375093965061110395505050505050565b34801561049c57600080fd5b50610287600435611189565b3480156104b457600080fd5b506103f3611250565b3480156104c957600080fd5b506040805180820182526102ef91369160049160449190839060029083908390808284375093965061126c95505050505050565b34801561050957600080fd5b506102ef6112b0565b34801561051e57600080fd5b506105276112b6565b604080517fffffffff000000000000000000000000000000000000000000000000000000009092168252519081900360200190f35b34801561056857600080fd5b506102ef6004356112da565b34801561058057600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff600435166024356112ef565b3480156105b157600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff600435811690602435166044356113a2565b3480156105e857600080fd5b506102c373ffffffffffffffffffffffffffffffffffffffff6004351661149d565b34801561061657600080fd5b506102ef73ffffffffffffffffffffffffffffffffffffffff600435166024356114c8565b34801561064757600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff60043516602435611522565b34801561067857600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff600435811690602435166044356115b6565b3480156106af57600080fd5b506102c373ffffffffffffffffffffffffffffffffffffffff60043516611610565b3480156106dd57600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff60043516611625565b34801561070b57600080fd5b506102c36004356116b2565b34801561072357600080fd5b506102ef6004356116dc565b34801561073b57600080fd5b506102ef611711565b34801561075057600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff60043516611717565b34801561077e57600080fd5b506103f361173e565b34801561079357600080fd5b5061079f60043561175a565b6040518082600260200280838360005b838110156107c75781810151838201526020016107af565b5050505090500191505060405180910390f35b3480156107e657600080fd5b506103f36004356117cf565b3480156107fe57600080fd5b50610287600435611806565b34801561081657600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff6004351661187a565b34801561084457600080fd5b50610850600435611925565b6040805194151585526020850193909352838301919091526060830152519081900360800190f35b34801561088457600080fd5b506102ef73ffffffffffffffffffffffffffffffffffffffff60043516611950565b3480156108b257600080fd5b5061028761199d565b3480156108c757600080fd5b506102876004356024351515611a30565b3480156108e457600080fd5b506108ed611ad0565b604080519687526020870195909552858501939093526060850191909152608084015260a0830152519081900360c00190f35b34801561092c57600080fd5b506040805160206004803580820135838102808601850190965280855261028795369593946024949385019291829185019084908082843750506040805187358901803560208181028481018201909552818452989b9a9989019892975090820195509350839250850190849080828437505060408051873589018035602081810284018101909452808352979a999883019791965082019450925082915060009085015b82821015610a0a5760408051808201825290808402870190600290839083908082843750505091835250506001909101906020016109d1565b50505050509192919290803590602001908201803590602001908080602002602001604051908101604052809392919081815260200183836020028082843750949750611b0d9650505050505050565b348015610a6657600080fd5b506103f3611c39565b348015610a7b57600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff60043516611c55565b348015610aa957600080fd5b5061028760043560243560443560643560843560a435611ce2565b348015610ad057600080fd5b506103f3611d61565b348015610ae557600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff60043516611d7d565b348015610b1357600080fd5b50610366611e0a565b348015610b2857600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff600435166024351515611e89565b348015610b5b57600080fd5b50610287600435602435611f45565b348015610b7657600080fd5b506103f3611fbc565b348015610b8b57600080fd5b50604080516020601f6064356004818101359283018490048402850184019095528184526102879473ffffffffffffffffffffffffffffffffffffffff8135811695602480359092169560443595369560849401918190840183828082843750949750611fd89650505050505050565b348015610c0757600080fd5b506102ef600435612039565b348015610c1f57600080fd5b506102c373ffffffffffffffffffffffffffffffffffffffff600435811690602435906044351661204e565b348015610c5757600080fd5b50610366600435612156565b348015610c6f57600080fd5b50610287600435602435612275565b348015610c8a57600080fd5b506102876004356024356122ec565b348015610ca557600080fd5b506102c373ffffffffffffffffffffffffffffffffffffffff60043581169060243516612363565b348015610cd957600080fd5b5061028773ffffffffffffffffffffffffffffffffffffffff6004351661239e565b7fffffffff000000000000000000000000000000000000000000000000000000001660009081526020819052604090205460ff1690565b60009081526017602052604090206004015490565b60155460009073ffffffffffffffffffffffffffffffffffffffff16331480610d875750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b80610da15750336000908152600c602052604090205460ff165b1515610dac57600080fd5b604080517f70a08231000000000000000000000000000000000000000000000000000000008152306004820152905173ffffffffffffffffffffffffffffffffffffffff8416916370a082319160248083019260209291908290030181600087803b158015610e1a57600080fd5b505af1158015610e2e573d6000803e3d6000fd5b505050506040513d6020811015610e4457600080fd5b5051831115610e5257600080fd5b8173ffffffffffffffffffffffffffffffffffffffff1663a9059cbb85856040518363ffffffff167c0100000000000000000000000000000000000000000000000000000000028152600401808373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200182815260200192505050602060405180830381600087803b158015610ef557600080fd5b505af1158015610f09573d6000803e3d6000fd5b505050506040513d6020811015610f1f57600080fd5b5051949350505050565b60009081526017602052604090205460ff1690565b60058054604080516020601f60027fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff610100600188161502019095169490940493840181900481028201810190925282815260609390929091830182828015610fe85780601f10610fbd57610100808354040283529160200191610fe8565b820191906000526020600020905b815481529060010190602001808311610fcb57829003601f168201915b505050505090505b90565b60009081526002602052604090205473ffffffffffffffffffffffffffffffffffffffff1690565b6000611026826117cf565b905073ffffffffffffffffffffffffffffffffffffffff838116908216141561104e57600080fd5b3373ffffffffffffffffffffffffffffffffffffffff8216148061107757506110778133612363565b151561108257600080fd5b60008281526002602052604080822080547fffffffffffffffffffffffff00000000000000000000000000000000000000001673ffffffffffffffffffffffffffffffffffffffff87811691821790925591518593918516917f8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b92591a4505050565b60155473ffffffffffffffffffffffffffffffffffffffff163314806111405750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b8061115a5750336000908152600c602052604090205460ff165b151561116557600080fd5b6000828152601760205260409020611184906002908101908390612e12565b505050565b60155473ffffffffffffffffffffffffffffffffffffffff163314806111c65750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b806111e05750336000908152600c602052604090205460ff165b15156111eb57600080fd5b600081815260176020526040812080547fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0016815560018101829055906112346002830182612e80565b5060006004820181905560059091015561124d81611806565b50565b60155473ffffffffffffffffffffffffffffffffffffffff1681565b6000816040518082600260200280838360005b8381101561129757818101518382015260200161127f565b5050505090500191505060405180910390209050919050565b60095490565b7f01ffc9a70000000000000000000000000000000000000000000000000000000081565b60009081526017602052604090206005015490565b60155473ffffffffffffffffffffffffffffffffffffffff1633148061132c5750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b806113465750336000908152600c602052604090205460ff165b151561135157600080fd5b303181111561135f57600080fd5b60405173ffffffffffffffffffffffffffffffffffffffff83169082156108fc029083906000818181858888f19350505050158015611184573d6000803e3d6000fd5b806113ad33826123cb565b806113cf575060155473ffffffffffffffffffffffffffffffffffffffff1633145b15156113da57600080fd5b73ffffffffffffffffffffffffffffffffffffffff841615156113fc57600080fd5b73ffffffffffffffffffffffffffffffffffffffff8316151561141e57600080fd5b611428848361245e565b61143284836124ff565b61143c83836126a4565b818373ffffffffffffffffffffffffffffffffffffffff168573ffffffffffffffffffffffffffffffffffffffff167fddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef60405160405180910390a450505050565b73ffffffffffffffffffffffffffffffffffffffff166000908152600c602052604090205460ff1690565b60006114d383611950565b82106114de57600080fd5b73ffffffffffffffffffffffffffffffffffffffff8316600090815260076020526040902080548390811061150f57fe5b9060005260206000200154905092915050565b60155473ffffffffffffffffffffffffffffffffffffffff1633148061155f5750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b806115795750336000908152600c602052604090205460ff165b151561158457600080fd5b61158e82826126fa565b6016546115b29073ffffffffffffffffffffffffffffffffffffffff166001611e89565b5050565b806115c133826123cb565b806115e3575060155473ffffffffffffffffffffffffffffffffffffffff1633145b15156115ee57600080fd5b61160a8484846020604051908101604052806000815250611fd8565b50505050565b600c6020526000908152604090205460ff1681565b600d5473ffffffffffffffffffffffffffffffffffffffff16331461164957600080fd5b73ffffffffffffffffffffffffffffffffffffffff8116151561166b57600080fd5b600e80547fffffffffffffffffffffffff00000000000000000000000000000000000000001673ffffffffffffffffffffffffffffffffffffffff92909216919091179055565b60009081526001602052604090205473ffffffffffffffffffffffffffffffffffffffff16151590565b60006116e66112b0565b82106116f157600080fd5b60098054839081106116ff57fe5b90600052602060002001549050919050565b600f5481565b336000908152600c602052604090205460ff16151561173557600080fd5b61124d81612749565b600e5473ffffffffffffffffffffffffffffffffffffffff1690565b611762612e8e565b60008281526017602052604090819020815180830192839052916002918201919082845b8154640100000000027fffffffffffffffffffffffffffffffffffffffffffffffffffffffff000000001681526001909101906020018083116117865750505050509050919050565b60008181526001602052604081205473ffffffffffffffffffffffffffffffffffffffff1680151561180057600080fd5b92915050565b60155473ffffffffffffffffffffffffffffffffffffffff163314806118435750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b8061185d5750336000908152600c602052604090205460ff165b151561186857600080fd5b61124d611874826117cf565b826127fd565b336000908152600c602052604090205460ff16151561189857600080fd5b60405173ffffffffffffffffffffffffffffffffffffffff8216907f1a2dbeec1a0714342f862f53c671a69a6cb438bf5aa44fc01a7ccff6fbde85ef90600090a273ffffffffffffffffffffffffffffffffffffffff166000908152600c6020526040902080547fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff00169055565b601760205260009081526040902080546001820154600483015460059093015460ff90921692909184565b600073ffffffffffffffffffffffffffffffffffffffff8216151561197457600080fd5b5073ffffffffffffffffffffffffffffffffffffffff1660009081526003602052604090205490565b600d5473ffffffffffffffffffffffffffffffffffffffff1633146119c157600080fd5b600d5460405173ffffffffffffffffffffffffffffffffffffffff909116907ff8df31144d9c2f0f6b59d69b8b98abd5459d07f2742c4df920b25aae33c6482090600090a2600d80547fffffffffffffffffffffffff0000000000000000000000000000000000000000169055565b60155473ffffffffffffffffffffffffffffffffffffffff16331480611a6d5750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b80611a875750336000908152600c602052604090205460ff165b1515611a9257600080fd5b60009182526017602052604090912080547fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0016911515919091179055565b600080600080600080600f5460106000600581101515611aec57fe5b0154601154601254601354601454949b939a50919850965094509092509050565b6000806000611b1a612e8e565b336000908152600c602052604081205460ff161515611b3857600080fd5b87518951148015611b4a575086518851145b8015611b57575085518751145b1515611b6257600080fd5b600094505b8851851015611c2e578885815181101515611b7e57fe5b9060200190602002015193508785815181101515611b9857fe5b9060200190602002015192508685815181101515611bb257fe5b9060200190602002015191508585815181101515611bcc57fe5b906020019060200201519050611be28383611103565b6000811115611bf557611bf58382612275565b611bff84846126fa565b601654611c239073ffffffffffffffffffffffffffffffffffffffff166001611e89565b600190940193611b67565b505050505050505050565b600e5473ffffffffffffffffffffffffffffffffffffffff1681565b600d5473ffffffffffffffffffffffffffffffffffffffff163314611c7957600080fd5b73ffffffffffffffffffffffffffffffffffffffff81161515611c9b57600080fd5b601580547fffffffffffffffffffffffff00000000000000000000000000000000000000001673ffffffffffffffffffffffffffffffffffffffff92909216919091179055565b60155473ffffffffffffffffffffffffffffffffffffffff16331480611d1f5750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b80611d395750336000908152600c602052604090205460ff165b1515611d4457600080fd5b600f95909555601093909355601191909155601255601355601455565b600d5473ffffffffffffffffffffffffffffffffffffffff1681565b600d5473ffffffffffffffffffffffffffffffffffffffff163314611da157600080fd5b73ffffffffffffffffffffffffffffffffffffffff81161515611dc357600080fd5b601680547fffffffffffffffffffffffff00000000000000000000000000000000000000001673ffffffffffffffffffffffffffffffffffffffff92909216919091179055565b60068054604080516020601f60027fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff610100600188161502019095169490940493840181900481028201810190925282815260609390929091830182828015610fe85780601f10610fbd57610100808354040283529160200191610fe8565b73ffffffffffffffffffffffffffffffffffffffff8216331415611eac57600080fd5b33600081815260046020908152604080832073ffffffffffffffffffffffffffffffffffffffff87168085529083529281902080547fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0016861515908117909155815190815290519293927f17307eab39ab6107e8899845ad3d59bd9653f200f220920489ca2b5937696c31929181900390910190a35050565b60155473ffffffffffffffffffffffffffffffffffffffff16331480611f825750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b80611f9c5750336000908152600c602052604090205460ff165b1515611fa757600080fd5b60009182526017602052604090912060050155565b60165473ffffffffffffffffffffffffffffffffffffffff1681565b81611fe333826123cb565b80612005575060155473ffffffffffffffffffffffffffffffffffffffff1633145b151561201057600080fd5b61201b8585856113a2565b61202785858585612933565b151561203257600080fd5b5050505050565b60009081526017602052604090206001015490565b60155460009073ffffffffffffffffffffffffffffffffffffffff1633148061208e5750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b806120a85750336000908152600c602052604090205460ff165b15156120b357600080fd5b8173ffffffffffffffffffffffffffffffffffffffff1663095ea7b385856040518363ffffffff167c0100000000000000000000000000000000000000000000000000000000028152600401808373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200182815260200192505050602060405180830381600087803b158015610ef557600080fd5b600e54604080517fc87b56dd00000000000000000000000000000000000000000000000000000000815260048101849052905160609273ffffffffffffffffffffffffffffffffffffffff169163c87b56dd91602480830192600092919082900301818387803b1580156121c957600080fd5b505af11580156121dd573d6000803e3d6000fd5b505050506040513d6000823e601f3d9081017fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe0168201604052602081101561222457600080fd5b81019080805164010000000081111561223c57600080fd5b8201602081018481111561224f57600080fd5b815164010000000081118282018710171561226957600080fd5b50909695505050505050565b60155473ffffffffffffffffffffffffffffffffffffffff163314806122b25750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b806122cc5750336000908152600c602052604090205460ff165b15156122d757600080fd5b60009182526017602052604090912060010155565b60155473ffffffffffffffffffffffffffffffffffffffff163314806123295750600d5473ffffffffffffffffffffffffffffffffffffffff1633145b806123435750336000908152600c602052604090205460ff165b151561234e57600080fd5b60009182526017602052604090912060040155565b73ffffffffffffffffffffffffffffffffffffffff918216600090815260046020908152604080832093909416825291909152205460ff1690565b600d5473ffffffffffffffffffffffffffffffffffffffff1633146123c257600080fd5b61124d81612ad2565b6000806123d7836117cf565b90508073ffffffffffffffffffffffffffffffffffffffff168473ffffffffffffffffffffffffffffffffffffffff16148061244657508373ffffffffffffffffffffffffffffffffffffffff1661242e84610ff3565b73ffffffffffffffffffffffffffffffffffffffff16145b8061245657506124568185612363565b949350505050565b8173ffffffffffffffffffffffffffffffffffffffff1661247e826117cf565b73ffffffffffffffffffffffffffffffffffffffff161461249e57600080fd5b60008181526002602052604090205473ffffffffffffffffffffffffffffffffffffffff16156115b257600090815260026020526040902080547fffffffffffffffffffffffff000000000000000000000000000000000000000016905550565b600080600061250e8585612b82565b60008481526008602090815260408083205473ffffffffffffffffffffffffffffffffffffffff8916845260079092529091205490935061255690600163ffffffff612c5716565b73ffffffffffffffffffffffffffffffffffffffff861660009081526007602052604090208054919350908390811061258b57fe5b9060005260206000200154905080600760008773ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020848154811015156125e557fe5b600091825260208083209091019290925573ffffffffffffffffffffffffffffffffffffffff8716815260079091526040812080548490811061262457fe5b600091825260208083209091019290925573ffffffffffffffffffffffffffffffffffffffff87168152600790915260409020805490612686907fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff8301612ea9565b50600093845260086020526040808520859055908452909220555050565b60006126b08383612c69565b5073ffffffffffffffffffffffffffffffffffffffff9091166000908152600760209081526040808320805460018101825590845282842081018590559383526008909152902055565b6127048282612d2b565b600980546000838152600a60205260408120829055600182018355919091527f6e1540171b6c0c960b71a7020d9f60077f6af931a8bbf590da0223dacf75c7af015550565b73ffffffffffffffffffffffffffffffffffffffff8116151561276b57600080fd5b60405173ffffffffffffffffffffffffffffffffffffffff82169033907f2931ebb3d190545dcf6801c37aa686b74f2e1000e753d0fac6e471a2aa5a621390600090a373ffffffffffffffffffffffffffffffffffffffff166000908152600c6020526040902080547fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff00166001179055565b600080600061280c8585612da0565b6000848152600b602052604090205460027fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff610100600184161502019091160415612868576000848152600b6020526040812061286891612ecd565b6000848152600a602052604090205460095490935061288e90600163ffffffff612c5716565b915060098281548110151561289f57fe5b90600052602060002001549050806009848154811015156128bc57fe5b600091825260208220019190915560098054849081106128d857fe5b6000918252602090912001556009805490612915907fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff8301612ea9565b506000938452600a6020526040808520859055908452909220555050565b6000806129558573ffffffffffffffffffffffffffffffffffffffff16612dfd565b15156129645760019150612ac9565b6040517f150b7a02000000000000000000000000000000000000000000000000000000008152336004820181815273ffffffffffffffffffffffffffffffffffffffff898116602485015260448401889052608060648501908152875160848601528751918a169463150b7a0294938c938b938b93909160a490910190602085019080838360005b83811015612a045781810151838201526020016129ec565b50505050905090810190601f168015612a315780820380516001836020036101000a031916815260200191505b5095505050505050602060405180830381600087803b158015612a5357600080fd5b505af1158015612a67573d6000803e3d6000fd5b505050506040513d6020811015612a7d57600080fd5b50517fffffffff0000000000000000000000000000000000000000000000000000000081167f150b7a020000000000000000000000000000000000000000000000000000000014925090505b50949350505050565b73ffffffffffffffffffffffffffffffffffffffff81161515612af457600080fd5b600d5460405173ffffffffffffffffffffffffffffffffffffffff8084169216907f8be0079c531659141344cd1fd0a4f28419497f9722a3daafe3b4186f6b6457e090600090a3600d80547fffffffffffffffffffffffff00000000000000000000000000000000000000001673ffffffffffffffffffffffffffffffffffffffff92909216919091179055565b8173ffffffffffffffffffffffffffffffffffffffff16612ba2826117cf565b73ffffffffffffffffffffffffffffffffffffffff1614612bc257600080fd5b73ffffffffffffffffffffffffffffffffffffffff8216600090815260036020526040902054612bf990600163ffffffff612c5716565b73ffffffffffffffffffffffffffffffffffffffff90921660009081526003602090815260408083209490945591815260019091522080547fffffffffffffffffffffffff0000000000000000000000000000000000000000169055565b600082821115612c6357fe5b50900390565b60008181526001602052604090205473ffffffffffffffffffffffffffffffffffffffff1615612c9857600080fd5b600081815260016020818152604080842080547fffffffffffffffffffffffff00000000000000000000000000000000000000001673ffffffffffffffffffffffffffffffffffffffff88169081179091558452600390915290912054612cfe91612e05565b73ffffffffffffffffffffffffffffffffffffffff90921660009081526003602052604090209190915550565b73ffffffffffffffffffffffffffffffffffffffff82161515612d4d57600080fd5b612d5782826126a4565b604051819073ffffffffffffffffffffffffffffffffffffffff8416906000907fddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef908290a45050565b612daa828261245e565b612db482826124ff565b604051819060009073ffffffffffffffffffffffffffffffffffffffff8516907fddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef908390a45050565b6000903b1190565b8181018281101561180057fe5b8260028101928215612e70579160200282015b82811115612e7057825182547fffffffff0000000000000000000000000000000000000000000000000000000016640100000000909104178255602090920191600190910190612e25565b50612e7c929150612f11565b5090565b506000815560010160009055565b60408051808201825290600290829080388339509192915050565b81548183558181111561118457600083815260209020611184918101908301612f4d565b50805460018160011615610100020316600290046000825580601f10612ef3575061124d565b601f01602090049060005260206000209081019061124d9190612f4d565b610ff091905b80821115612e7c5780547fffffffff00000000000000000000000000000000000000000000000000000000168155600101612f17565b610ff091905b80821115612e7c5760008155600101612f535600a165627a7a723058207f0ddc1b20db5fac54021c30d0f91c32d8804bd7ba0f4c4f307b58b97a9be2910029000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000007436c6f76657273000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004434c565200000000000000000000000000000000000000000000000000000000",
			"gas": "0x3527c1",
			"gasPrice": "0x0"
		}
	]
}
```
However when I trigger this tx manually the results seem to be correct.
```
{
  "jsonrpc": "2.0",
  "id": 20,
  "result": "0xd1fa3738a90e190c36585b84ee4ee6778f0e464d73c3d178c467eb5a23adf792"
}
```
I noticed here https://github.com/ChainSafe/ethermint/blob/development/rpc/types.go#L42 a todo with this error appearing. I'm not sure why I'm getting the error but would it be safe to assume if this todo were completed it would not occur?

**Comment by austinabell** [comment]
this is intended, Cosmos/Tendermint does not process the transactions until the end of the block, the messages are not processed. Until the transactions can be processed to essentially simulate the pending state through Tendermint, this will most likely stay as such. 

To calculate and save the pending state becomes very clunky and inconsistent quickly, so unless absolutely needed, it probably shouldn't be done. If pending state isn't necessarily needed, you can just point the pending tag to "latest" and it won't error. Let me know if you do depend on the pending state for queries.

I can update the error message and comment if this is confusing, but I don't think it's a good idea to change the "pending" query tag to "latest" as default.

**Comment by okwme** [comment]
I think it has to do with the truffle/web3 library deploying a contract and then querying for the pending block. Not sure if returning the latest block number would break the flow of deploying contracts this way. I'll change the function to return latest instead of an error and see if the deploy works.

**Comment by okwme** [comment]
Changing the results of pending to `EarliestBlockNumber` or `LatestBlockNumber` doesn't work. When I try running a more standard truffle project (similar to what aragon uses) the error returned is:
```
{ code: -32000,
  message:
   'Block: response error: RPC error -32603 - Internal error: Height must be less than or equal to the current blockchain height' }
```
I'm not sure I understand the problem with returning a pending block with tendermint. Some content of the return statement isn't available until the block is completed (at which time it is no longer pending?). If that's the case maybe it's ok to return a block that just doesn't return a hash, state root or size but has all the other information?
```
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "difficulty": null,
    "extraData": null,
    "gasLimit": "0xffffffff",
    "gasUsed": "0x0",
    "hash": null,
    "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "miner": "0x0000000000000000000000000000000000000000",
    "nonce": null,
    "number": "0x1870",
    "parentHash": "0x7d5cd6928c923ecf4962c072e4304ab7c77d71b5ba24242fe8dab06b9d86a83c",
    "sha3Uncles": null,
    "size": "0x229",
    "stateRoot": null",
    "timestamp": "0x5dd066a0",
    "totalDifficulty": null,
    "transactions": [],
    "transactionsRoot": "0x",
    "uncles": null
  }
}
```

**Comment by austinabell** [comment]
The issue is that the messages in Cosmos aren't handled until the end of the block, which is when the evm state transition happens. Ethereum is different because it applies each state transition and builds the pending block before the end of the block. To do something similar there would have to be something that has to happen in the AnteHandler, and I don't think this will work out cleanly if possible.

I'll look into this in more detail later but when I tested truffle last I was able to deploy multiple connected contracts without an issue.
